modules/correlation/id-counter.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 OneIdentity LLC.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "id-counter.h"
+
+IdCounter *id_counter_new(void)
+{
+  IdCounter *self = g_new(IdCounter, 1);
+  self->id = 0;
+  self->ref_count = 1;
+  return self;
+}
+
+guint id_counter_get_next_id(IdCounter *self)
```

Comment:
Unfortunately we cannot enforce this by our style checker because of a limitation of `astyle`, we tend to separate the return type and the function name to separate lines in the `.c` files. The reason is that we can find implementations a lot easier with grep if we do an anchored regexp, e.g.:
```
$ git grep "^log_threaded_dest_driver_init_instance"
lib/logthrdest/logthrdestdrv.c:log_threaded_dest_driver_init_instance(LogThreadedDestDriver *self, GlobalConfig *cfg)
```

Could you please do this for the new functions in this file? Thanks!